xds/credentials/cert_manager.go

```diff
@@ -0,0 +1,271 @@
+}
+
+//GetCertificate return certificate of application
+func (c *AgentCertManager) GetCertificate() ([]tls.Certificate, error) {
+	material, err := c.provider.KeyMaterial(context.Background())
+	if err != nil {
+		return nil, err
+	}
+	return material.Certs, nil
+}
+
+// buildProvider  build cert provider from config
+func buildProvider(configs map[string]*certprovider.BuildableConfig, instanceName string) (certprovider.Provider, error) {
+	cfg, ok := configs[instanceName]
+	if !ok {
+		return nil, fmt.Errorf("certificate provider instance %q not found in bootstrap file", instanceName)
+	}
+	provider, err := cfg.Build(certprovider.BuildOptions{
+		CertName:     "ca",
+		WantIdentity: true,
+		WantRoot:     true,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("xds: failed to get security plugin instance (%+v): %v", cfg, err)
+	}
+	return provider, nil
+}
+
+// CACertManager work in no agent mode, fetch cert form CA
+type CACertManager struct {
+	// Certs contains a slice of cert/key pairs used to prove local identity.
+	Certs []tls.Certificate
+	// Roots contains the set of trusted roots to validate the peer's identity.
+	Roots *x509.CertPool
+
+	NoAfter time.Time
+
+	RootNoAfter time.Time
+
+	rootPath string
+}
+
+//GetCertificate return certificate of application
+func (c *CACertManager) GetCertificate() ([]tls.Certificate, error) {
+	//cert expired
+	if time.Now().After(c.NoAfter) {
+		if err := c.UpdateCert(); err != nil {
+			return nil, err
+		}
+	}
+	return c.Certs, nil
+}
+
+//GetRootCertificate return certificate of ca
+func (c *CACertManager) GetRootCertificate() (*x509.CertPool, error) {
+	//root expired
+	if time.Now().After(c.RootNoAfter) {
+		if err := c.UpdateRoot(); err != nil {
+			return nil, err
+		}
+	}
+	return c.Roots, nil
+}
+
+// UpdateRoot update root cert
+func (c *CACertManager) UpdateRoot() error {
+	rootFileContents, err := ioutil.ReadFile(c.rootPath)
+	if err != nil {
+		return err
+	}
+	trustPool := x509.NewCertPool()
+	if !trustPool.AppendCertsFromPEM(rootFileContents) {
+		logger.Warn("failed to parse root certificate")
+	}
+	c.Roots = trustPool
+	block, _ := pem.Decode(rootFileContents)
+	if block == nil {
+		return fmt.Errorf("failed to decode certificate")
+	}
+	cert, err := x509.ParseCertificate(block.Bytes)
+	if err != nil {
+		return nil
+	}
+	c.RootNoAfter = cert.NotAfter
+	return nil
+}
+
+// UpdateCert update  cert
+func (c *CACertManager) UpdateCert() error {
+	tokenProvider, err := NewSaTokenProvider(ServiceAccountPath)
+	if err != nil {
+		return err
+	}
+
+	trustRoot, err := c.GetRootCertificate()
+	if err != nil {
+		return nil
+	}
+	citadelClient, err := remote.NewCitadelClient(&remote.Options{
+		CAEndpoint:    IstioCAEndpoint,
+		TrustedRoots:  trustRoot,
+		TokenProvider: tokenProvider,
+		CertSigner:    certSigner,
+		ClusterID:     clusterID,
+	})
+	if err != nil {
+		return err
+	}
+	host := "spiffe://" + "cluster.local" + "/ns/" + PodNamespace + "/sa/" + "default"
```

Comment:
![image](https://user-images.githubusercontent.com/10649416/188546010-9252c42e-32b2-4711-b12d-da0f14efaef6.png)

Could we make this host configurable, like this `URIPrifix`?